core/dbt/include/global_project/macros/materializations/helpers.sql

```diff
@@ -72,3 +72,96 @@
   {% endif %}
   {% do return(config_full_refresh) %}
 {% endmacro %}
+
+
+{% macro incremental_validate_on_schema_change(on_schema_change, default_value='ignore') %}
+   
+   {% if on_schema_change not in ['sync', 'append', 'fail', 'ignore'] %}
+     {{ return(default_value) }}
+
+   {% else %}
+     {{ return(on_schema_change) }}
+   
+   {% endif %}
+
+{% endmacro %}
+
+{% macro get_column_names(columns) %}
+  {% set result = [] %}
+  
+  {% for col in columns %}
+    {{ result.append(col.column) }}
+  {% endfor %}
+  
+  {{ return(result) }}
+{% endmacro %}
+
+{% macro diff_columns(source_columns, target_columns) %}
```

Comment:
I just searched for "Jinja set difference" and, of course, came across https://github.com/fishtown-analytics/dbt/issues/1894 :)

I do wonder if something like is better implemented as a python method. Let's think about the right balance between clarity and confusion. Python obfuscates logic away from user space, but it can express certain logical operations much more cleanly.

Reply:
Yeah I was a bit annoyed to have to write this hehe. This approach seems relatively low risk for pushing the enhancement to incremental logic forward, and I'd be happy to dive into adding a python method to enhance the concision in the coming weeks.